tests/unit/test_grpc_index.py

```diff
@@ -0,0 +1,239 @@
+        self.md1 = {'genre': 'action', 'year': 2021}
+        self.md2 = {'genre': 'documentary', 'year': 2020}
+        self.filter1 = {'genre': {'$in': ['action']}}
+        self.filter2 = {'year': {'$eq': 2020}}
+
+        pinecone.init(api_key='example-key')
+        self.index = pinecone.GRPCIndex('example-name')
+
+    # region: upsert tests
+
+    def test_upsert_tuplesOfIdVec_UpserWithoutMD(self, mocker):
+        mocker.patch.object(self.index, '_wrap_grpc_call', autospec=True)
+        self.index.upsert([('vec1', self.vals1), ('vec2', self.vals2)], namespace='ns')
+        self.index._wrap_grpc_call.assert_called_once_with(
+            self.index.stub.Upsert,
+            UpsertRequest(
+                vectors=[
+                    Vector(id='vec1', values=self.vals1, metadata={}),
+                    Vector(id='vec2', values=self.vals2, metadata={})],
+                namespace='ns'),
+            timeout=None
+        )
+
+    def test_upsert_tuplesOfIdVecMD_UpsertVectorsWithMD(self, mocker):
+        mocker.patch.object(self.index, '_wrap_grpc_call', autospec=True)
+        self.index.upsert([('vec1', self.vals1, self.md1), ('vec2', self.vals2, self.md2)], namespace='ns')
+        self.index._wrap_grpc_call.assert_called_once_with(
+            self.index.stub.Upsert,
+            UpsertRequest(
+                vectors=[
+                    Vector(id='vec1', values=self.vals1, metadata=dict_to_proto_struct(self.md1)),
+                    Vector(id='vec2', values=self.vals2, metadata=dict_to_proto_struct(self.md2))],
+                namespace='ns'),
+            timeout=None)
+
+    def test_upsert_vectors_upsertInputVectors(self, mocker):
+        mocker.patch.object(self.index, '_wrap_grpc_call', autospec=True)
+        self.index.upsert([Vector(id='vec1', values=self.vals1, metadata=dict_to_proto_struct(self.md1)),
+                           Vector(id='vec2', values=self.vals2, metadata=dict_to_proto_struct(self.md2))],
+                          namespace='ns')
+        self.index._wrap_grpc_call.assert_called_once_with(
+            self.index.stub.Upsert,
+            UpsertRequest(
+                vectors=[
+                    Vector(id='vec1', values=self.vals1, metadata=dict_to_proto_struct(self.md1)),
+                    Vector(id='vec2', values=self.vals2, metadata=dict_to_proto_struct(self.md2))],
+                namespace='ns'),
+            timeout=None)
+
+    # endregion
+
+    # region: query tests
+
+    def test_query_byVectorNoFilter_queryVectorNoFilter(self, mocker):
+        mocker.patch.object(self.index, '_wrap_grpc_call', autospec=True)
+        self.index.query(top_k=10, vector=self.vals1)
+        self.index._wrap_grpc_call.assert_called_once_with(
+            self.index.stub.Query,
+            QueryRequest(top_k=10, vector=self.vals1),
+            timeout=None,
+        )
+
+    def test_query_byVectorWithFilter_queryVectorWithFilter(self, mocker):
```

Comment:
Consider checking also the return value, to make sure the `parse_query_response()` part works well

Reply:
Deferred to [backlog](https://app.asana.com/0/1203260648987893/1203612700548377), resolving